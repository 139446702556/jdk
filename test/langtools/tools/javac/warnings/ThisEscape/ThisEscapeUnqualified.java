/*
 * @test /nodynamiccopyright/
 * @bug 8015831
 * @compile/ref=ThisEscapeUnqualified.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeUnqualified.java
 * @summary Verify proper 'this' escape interpretation of unqualified non-static method invocations.
 */

public class ThisEscapeUnqualified {

    // This class has a leak
    public static class Example1 {

        public Example1() {
            new Inner();
        }

        public final class Inner {
            public Inner() {
                mightLeak();    // refers to Example1.mightLeak()
            }
        }

        public void mightLeak() {
        }
    }

    // This class does NOT have a leak
    public static class Example2 {

        public Example2() {
            new Inner();
        }

        public final class Inner {
            public Inner() {
                mightLeak();    // refers to Inner.mightLeak()
            }

            public void mightLeak() {
            }
        }

        public void mightLeak() {
        }
    }
}
