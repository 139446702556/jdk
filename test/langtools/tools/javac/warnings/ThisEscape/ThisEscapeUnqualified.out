ThisEscapeUnqualified.java:14:13: compiler.warn.possible.this.escape
ThisEscapeUnqualified.java:19:26: compiler.warn.possible.this.escape.location
2 warnings
